src/Service.Tests/SqlTests/RestApiTests/RestBodyFlexibleModeTests/RestBodyNonStrictModeTests.cs

```diff
@@ -0,0 +1,180 @@
+using Azure.DataApiBuilder.Service.Exceptions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Azure.DataApiBuilder.Service.Tests.SqlTests.RestApiTests
+{
+    /// <summary>
+    /// Class containing integration tests to validate scenarios when we operate in non-strict mode for REST request body,
+    /// i.e. we allow extraneous fields to be present in the request body.
+    /// </summary>
+    public abstract class RestBodyNonStrictModeTests : RestApiTestBase
+    {
+        #region Positive tests
+
+        /// <summary>
+        /// Test to validate that extraneous fields are allowed in request body when we operate in runtime.rest.request-body-strict = false.
+        /// </summary>
+        [TestMethod]
+        public virtual async Task MutationsTestWithExtraneousFieldsInRequestBody()
+        {
+            // Validate that when a field which does not map to any column (here 'non_existing_field')
+            // in the underlying table is included in the request body for POST operation,
+            // it is just ignored when we do not operate in strict mode for REST.
+            string requestBody = @"
+            {
+                ""categoryid"": ""3"",
+                ""pieceid"": ""1"",
+                ""piecesAvailable"": null,
+                ""piecesRequired"": 1,
+                ""categoryName"":""SciFi"",
+                ""non_existing_field"": 5
+            }";
+
+            await SetupAndRunRestApiTest(
+                    primaryKeyRoute: null,
+                    queryString: null,
+                    entityNameOrPath: _Composite_NonAutoGenPK_EntityPath,
+                    sqlQuery: GetQuery("InsertOneWithNonExistingFieldInRequestBody"),
+                    operationType: EntityActionOperation.Insert,
+                    requestBody: requestBody,
+                    expectedStatusCode: HttpStatusCode.Created,
+                    expectedLocationHeader: "categoryid/3/pieceid/1"
+                );
+
+            // Validate that when PK fields (here 'categoryid' and 'pieceid') are included in the request body for PUT operation,
+            // they are just ignored when we do not operate in strict mode for REST. The values of the fields from the URL is used.
+            requestBody = @"
+            {
+               ""categoryName"":""SciFi"",
+               ""piecesAvailable"":""10"",
+               ""piecesRequired"":""5"",
+               ""piecied"": 5,
+               ""categoryid"": 4
+            }";
+
+            await SetupAndRunRestApiTest(
+                primaryKeyRoute: "categoryid/2/pieceid/1",
+                queryString: null,
+                entityNameOrPath: _Composite_NonAutoGenPK_EntityPath,
+                sqlQuery: GetQuery("PutOneWithPKFieldsInRequestBody"),
+                operationType: EntityActionOperation.Upsert,
+                requestBody: requestBody,
+                expectedStatusCode: HttpStatusCode.OK
+                );
```

Comment:
I like this test, and want to see Jerry's perspective on whether we should care that PK values in URI differ from PK values in the request body. We ignore request body PK's and their values right now. If for whatever reason that changes, maybe this would be a good candidate to be in a separate test method?

Reply:
I see he defined "Are ignored properties compared to the URL key value? Never. They are ignored."

Reply:
Yes, and we are implementing this very behavior.